### PR TITLE
stop using $root to make easyblock compatible with module files in Lua syntax

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/hpcugent/easybuild-framework/pull/1060

### CODE REVIEW

**Comment by pforai** [comment]
:+1: here as well!

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
@wpoely86: can you give this a thorough review? should be fine to go in, but it's good to have someone else double-check

File: easybuild/easyblocks/o/openbabel.py [hunk 1]
```diff
@@ -76,6 +76,8 @@ def make_module_extra(self):
         """Custom variables for OpenBabel module."""
         txt = super(EB_OpenBabel, self).make_module_extra()
         txt += self.module_generator.prepend_paths('PYTHONPATH', ['lib'])
-        txt += self.module_generator.set_environment('BABEL_LIBDIR', '$root/lib/openbabel/%s/' % self.version)
-        txt += self.module_generator.set_environment('BABEL_DATADIR', '$root/share/openbabel/%s/' % self.version)
+        babel_libdir = os.path.join(self.installdir, 'lib', 'openbabel', self.version)
+        txt += self.module_generator.set_environment('BABEL_LIBDIR', babel_libdir)
+        babel_datadir = os.path.join('share', 'openbabel', self.version)
```
Comment (wpoely86) [inline easybuild/easyblocks/o/openbabel.py]: `self.installdir,`

File: easybuild/easyblocks/p/python_meep.py [hunk 2]
```diff
@@ -136,10 +136,10 @@ def make_module_extra(self):
 
         meep = get_software_root('Meep')
```
Comment (wpoely86) [inline easybuild/easyblocks/p/python_meep.py]: why is this here? Why not `self.installdir`?

**Comment by wpoely86** [comment]
looks good

File: easybuild/easyblocks/p/python_meep.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/p/python_meep.py]: Meep != python-meep

File: easybuild/easyblocks/o/openbabel.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/o/openbabel.py]: wow, nice catch

**Comment by boegel** [comment]
I've tested this thoroughly by rebuilding all existing easyconfigs with the following toolchains: `goolf/1.4.10`, `ictce/5.3.0`, `intel/*` and `foss/*`, + NWChem with `ictce/4.1.13`

`eb` can now generate Lua module files for all of those builds, hooray!
